pkg/txn/storage/memorystorage/memorytable/tuple.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memorytable
+
+import (
+	"fmt"
+
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+)
+
+type Tuple []any
+
+var _ Ordered[Tuple] = Tuple{}
+
+func (t Tuple) Less(than Tuple) bool {
+	i := 0
+	for i < len(t) {
+		a := t[i]
+		if i >= len(than) {
+			return false
+		}
+		b := than[i]
+
+		// min, max
+		if a == Min {
+			return b != Min
+		}
+		if a == Max {
+			return false
+		}
+		if b == Min {
+			return false
+		}
+		if b == Max {
+			return a != Max
+		}
+
+		switch a := a.(type) {
+
```

Comment:
This type switch, we should try to hid this somewhere in types.    In theory (almost) all type switch of generic operations like assgin, compare, if ever needed should be in types package.

Reply:
container/types package doesn't define primitive types such as Int32, Bool. It's using go builtin types everywhere, hard to change that.